### PR TITLE
Clean-up history API request handlers

### PR DESCRIPTION
**Pull request by buddy326**

### CODE REVIEW

**Comment by buddy326** [comment]
Not sure if i should be raising `ResourceNotFoundError()` from history.data.get() or just return None lilke Im doing right now. Thats what the get() method was doing in history.db. i'm sure some things could get changed so just let me know how i can improve it.

**Comment by igboyes** [comment]
Yes. Raise the error instead of returning `None`.

Are you going to work on `revert()`?

**Comment by buddy326** [comment]
ok, and yes I worked on it a little yesterday but wanted clarification on the get() first too.

File: virtool/history/data.py [hunk 1]
```diff
@@ -1,38 +1,100 @@
...
+        """
+        document = await self._db.history.find_one(change_id, PROJECTION)
+
+        if document:
+            document = await apply_transforms(
+                virtool.utils.base_processor(document),
+                [AttachUserTransform(self._db), DiffTransform(self.data_path)],
+            )
+
+            return document
```
Comment (igboyes) [inline virtool/history/data.py:46]:
```suggestion
            return await apply_transforms(
                virtool.utils.base_processor(document),
                [AttachUserTransform(self._db), DiffTransform(self.data_path)],
            )
```

File: virtool/history/data.py [hunk 2]
```diff
@@ -1,38 +1,100 @@
...
+                for sequence in sequences:
+                    await self._db.sequences.insert_one(sequence)
+
+                # Replace the existing otu with the patched one. If it doesn't exist, insert it.
+                await self._db.otus.replace_one(
+                    {"_id": otu_id}, patched_otu, upsert=True
+                )
+
+            else:
+                await self._db.otus.delete_one({"_id": otu_id})
```
Comment (igboyes) [inline virtool/history/data.py:96]: Flip this so it is `if patched is None:`.

**Review by igboyes — CHANGES_REQUESTED** [review]
Just a few things. Excellent work!

**Review by igboyes — COMMENTED** [review]
I'll merge once you fix Deepsource issues.

**Review by igboyes — APPROVED** [review]

**Pull request merged**